### DISCUSSION TITLE
sending/adding  midi messages to the "inport" message queue

### DISCUSSION THREAD

**Discussion by rolfiepolfie** [mido/mido#427] [Q&A]
Not directly an issue , but let's say the usual: "**inport** = mido.open_input()"

I usually receive midi-messages from an external device , these will be intercepted in the **inport**.

What if I want to do the same in  code , using no external devices, and place midi-messages directly in the inport-message queue, so it only "looks" it came from an external device.
I guess I had to tap into the **inport's** message queue?
Is this possible? and is there a short description what I have to hack? 
thanks

**Comment by Mickinator** [comment]
Hi rolfiepolfie,
I use the queue for "serial data" like it comes from a keyboard.

Other massages (e.g. "all notes off") I send directly using

ano = Message("control_change", channel=0, control=123, value=0)
output.send(aso)

I hope this helps 🌈

**Comment by onlaj** [comment]
@Mickinator you are sending it to output which can't be (I'm not sure about that) iterate the same way as input.
@rolfiepolfie is trying to somehow inject message into input port

Something like this:

```
inport = mido.open_input("portname")
example_message = Message("control_change", channel=0, control=123, value=0)
inport.inject_message(example_message) # this function obviously doesn't exist
```

So when you iterate through messages pending in inport it will be there

```
pending_messages = inport.iter_pending()
for msg in midipending:
    print(msg)

# output: "control_change", channel=0, control=123, value=0
```

@rolfiepolfie Did you find a solution?

**Comment by rolfiepolfie** [comment]
> @Mickinator you are sending it to output which can't be (I'm not sure about that) iterate the same way as input.
> @rolfiepolfie is trying to somehow inject message into input port
> 
> Something like this:
> 
> ```
> inport = mido.open_input("portname")
> example_message = Message("control_change", channel=0, control=123, value=0)
> inport.inject_message(example_message) # this function obviously doesn't exist
> ```
> 
> So when you iterate through messages pending in inport it will be there
> 
> ```
> pending_messages = inport.iter_pending()
> for msg in midipending:
>     print(msg)
> 
> # output: "control_change", channel=0, control=123, value=0
> ```
> 
> @rolfiepolfie Did you find a solution?

Hello, and thanks for showing interest, your interpretation of my question was spot on. I simply avoided the problem and bought a little midi controller (like to sit and program in coffeehouses) it worked well for the task.
